Formula/youtube-dlc.rb

```diff
@@ -0,0 +1,37 @@
+class YoutubeDlc < Formula
+  desc "Media downloader supporting various sites such as youtube"
+  homepage "https://github.com/blackjack4494/yt-dlc"
+  url "https://github.com/blackjack4494/yt-dlc/archive/2020.10.31.tar.gz"
+  sha256 "0d23cc2e2f1cc7291e5d2a468b1cd282a6be228a215df188aaecae8951ac64d9"
+  license "Unlicense"
+
+  livecheck do
+    url "https://github.com/blackjack4494/yt-dlc/releases/latest"
+    regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+  end
```

Comment:
@samford @nandahkrishna  opinions?

Reply:
This is the default regex we use in case latest versions are tagged on GitHub, seems okay to me.

Reply:
Does that work with date based versions?

Reply:
In this case, the date-based version is of the form `20xx.xx.xx`, which is matched by this regex. So, yes this should work.

Looking at the tags/previous releases, there's been a `2020.10.24-6` which wouldn't be matched by this regex. Modifying the regex to `%r{href=.*?/tag/v?(\d+(?:\.\d+)+(?:-\d+)?)["' >]}I` would match that version too, but I'd wait for @samford's opinion on this.

Reply:
We may want to simply modify the delimiter between the numbers to allow for `.` or `-`, like `%r{href=.*?/tag/v?(\d+(?:[.-]\d+)+)["' >]}i`. This accounts for the `2020.10.24-6` format but would also match unexpected version formats like `2020.10.24.6`, `2020-10-24`, `2020-10-24-6`, etc.